funnel/views/errors.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import unicode_literals
+
+from flask import current_app, redirect, request
+from werkzeug.routing import RequestRedirect
+import werkzeug.exceptions as exceptions
+
+from coaster.views import render_with
+
+from .. import app
+from ..models import db
+
+
+@app.errorhandler(exceptions.Forbidden)
+@render_with('403.html.jinja2', json=True)
+def error403(e):
+    return {'error': "403 Forbidden"}, 403
+
+
+@app.errorhandler(exceptions.NotFound)
+@render_with('404.html.jinja2', json=True)
+def error404(e):
+    if request.path.endswith('/') and request.method == 'GET':
+        newpath = request.path[:-1]
+        try:
+            adapter = current_app.url_map.bind_to_environ(request)
+            matchinfo = adapter.match(newpath)
+            if matchinfo[0] != request.endpoint:
+                # Redirect only if it's not back to the same endpoint
+                redirect_url = request.base_url[:-1]
+                if request.query_string:
+                    redirect_url = (
+                        redirect_url + '?' + request.query_string.decode('utf-8')
+                    )
+                return redirect(redirect_url)
+        except (exceptions.NotFound, RequestRedirect, exceptions.MethodNotAllowed):
+            pass
+    return {'error': "404 Not Found"}, 404
```

Comment:
I _think_ we don't need to implement this method at all, because Baseframe's existing handler will pick up our new template. Try it?